packages/anoncreds/src/services/AnonCredsHolderServiceOptions.ts

```diff
@@ -0,0 +1,102 @@
+import { CredentialInfo, RequestedCredentials } from '../models'
+
+import type {
+  AnonCredsCredential,
+  AnonCredsCredentialOffer,
+  AnonCredsCredentialRequest,
+  AnonCredsProofRequest,
+  NonRevokedInterval,
+  ReferentWalletQuery,
+} from '../models/exchange'
+
+import type {
+  AnonCredsCredentialDefinition,
+  AnonCredsRevocationList,
+  AnonCredsRevocationRegistryDefinition,
+  AnonCredsSchema,
+} from '../models/vdr'
+
+export interface AttributeInfo {
+  name?: string
+  names?: string[]
+}
+
+export interface CreateProofOptions {
+  proofRequest: AnonCredsProofRequest
+  requestedCredentials: RequestedCredentials
+  schemas: {
+    [schemaId: string]: AnonCredsSchema
+  }
+  credentialDefinitions: {
+    [credentialDefinitionId: string]: AnonCredsCredentialDefinition
+  }
+  revocationStates: {
+    [revocationRegistryDefinitionId: string]: {
+      definition: AnonCredsRevocationRegistryDefinition
+      revocationLists: {
+        [timestamp: string]: AnonCredsRevocationList
+      }
+    }
+  }
+}
+
+export type CreateProofMetadata = {
+  schemasMetaData: { [key: string]: { seqNo: number; ver: string } }
+  credentialDefinitionsMetaData: { [key: string]: { ver: string } }
+}
+
+export interface StoreCredentialOptions {
+  // TODO: what is in credential request metadata?
+  credentialRequestMetadata: Record<string, unknown>
+  credential: AnonCredsCredential
+  credentialDefinition: AnonCredsCredentialDefinition
+  credentialId?: string
+  revocationRegistryDefinition?: AnonCredsRevocationRegistryDefinition
+}
+export type StoreCredentialMetadata = {
+  credentialDefinitionMetadata: {
+    id: string
+    ver: string
+  }
+  revocationRegistryDefinition: {
+    id: string
```

Comment:
For the create credential definition method we pass the schema with id as:

```
{
  value: schema,
   id: schemaId
}
```

As we're only including the id here, maybe we use that appraoch as well. The id doesn't feel like metadata to me

Reply:
Otherwise we should update the createCredentialDefinition method to also put the id in the metadata for consistency

Reply:
**consistency**